python/paddle/fluid/tests/unittests/test_pad3d_op.py

```diff
@@ -0,0 +1,387 @@
+        np_out = self._get_numpy_out(input_data, pad, mode, value)
+        place = paddle.CPUPlace()
+        with dg.guard(place) as g:
+            input = dg.to_variable(input_data)
+            output = F.pad(x=input, pad=pad, mode=mode, value=value)
+            self.assertTrue(np.allclose(output.numpy(), np_out))
+
+
+class TestPadAPI(unittest.TestCase):
+    def _get_numpy_out(self, input_data, pad, mode, value, data_format="NCDHW"):
+        if data_format == "NCDHW":
+            pad = [
+                (0, 0),
+                (0, 0),
+                (pad[4], pad[5]),
+                (pad[2], pad[3]),
+                (pad[0], pad[1]),
+            ]
+        else:
+            pad = [
+                (0, 0),
+                (pad[4], pad[5]),
+                (pad[2], pad[3]),
+                (pad[0], pad[1]),
+                (0, 0),
+            ]
+
+        if mode == "constant":
+            out = np.pad(input_data, pad, mode=mode, constant_values=value)
+        elif mode == "reflect":
+            out = np.pad(input_data, pad, mode=mode)
+        elif mode == "replicate":
+            out = np.pad(input_data, pad, mode="edge")
+        elif mode == "circular":
+            out = np.pad(input_data, pad, mode="wrap")
+
+        return out
+
+    def setUp(self):
+        self.places = [paddle.CPUPlace()]
+        if core.is_compiled_with_cuda():
+            self.places.append(paddle.CUDAPlace(0))
+
+    def check_static_result(self, place):
+        with program_guard(Program(), Program()):
+            input_shape = (1, 2, 3, 4, 5)
+            pad = [1, 2, 1, 1, 3, 4]
+            mode = "constant"
+            value = 100
+            input_data = np.random.rand(*input_shape).astype(np.float32)
+            x = paddle.data(name="x", shape=input_shape)
+            result = F.pad(x=x, pad=pad, value=value, mode='constant')
+            exe = Executor(place)
+            fetches = exe.run(default_main_program(),
+                              feed={"x": input_data},
+                              fetch_list=[result])
+
+            np_out = self._get_numpy_out(input_data, pad, mode, value)
+
+            self.assertTrue(np.allclose(fetches[0], np_out))
+
+    def test_static(self):
+        for place in self.places:
+            self.check_static_result(place=place)
+
+    def test_dygraph(self):
+        for place in self.places:
+            input_shape = (1, 2, 3, 4, 5)
+            pad = [1, 2, 1, 1, 3, 4]
+            mode = "constant"
+            value = 100
+            input_data = np.random.rand(*input_shape).astype(np.float32)
+            np_out = self._get_numpy_out(input_data, pad, mode, value)
+            with dg.guard(place) as g:
+                input = dg.to_variable(input_data)
+                output = F.pad(x=input, pad=pad, mode=mode, value=value)
+                self.assertTrue(np.allclose(output.numpy(), np_out))
+
+
+class TestPad1dClass(unittest.TestCase):
```

Comment:
what happens to 2d? also, why is a separate test class needed?

Reply:
Add Test1d 2 and 3d api separately, thx